src/main/java/org/elasticsearch/common/lucene/search/function/FunctionScoreQuery.java

```diff
@@ -171,12 +181,12 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass())
             return false;
         FunctionScoreQuery other = (FunctionScoreQuery) o;
-        return this.getBoost() == other.getBoost() && this.subQuery.equals(other.subQuery) && this.function.equals(other.function)
+        return this.getBoost() == other.getBoost() && this.subQuery.equals(other.subQuery) && (this.function != null ? this.function.equals(other.function) : other.function == null)
                 && this.maxBoost == other.maxBoost;
     }
 
     @Override
     public int hashCode() {
-        return subQuery.hashCode() + 31 * function.hashCode() ^ Float.floatToIntBits(getBoost());
+        return subQuery.hashCode() + 31 * (function == null ? 0 : function.hashCode()) ^ Float.floatToIntBits(getBoost());
```

Comment:
Note that you can use Objects.hashCode(function) directly which will make sure to return 0 if the value is null.